File: app/src/main/java/com/nathanhaze/gifcreator/gallery/PhotoGallery.java
package com.nathanhaze.gifcreator.gallery;


import android.app.Fragment;
import android.os.Bundle;
import android.view.MenuItem;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.NavUtils;
import androidx.fragment.app.FragmentTransaction;

import com.nathanhaze.gifcreator.R;

/**
 * A simple {@link Fragment} subclass.
 */
public class PhotoGallery extends AppCompatActivity {

    public PhotoGallery() {
        // Required empty public constructor
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_photo_gallery);
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.frame_main, new GalleryFragment());
        ft.commit();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                NavUtils.navigateUpFromSameTask(this);
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/com/nathanhaze/gifcreator/gallery/PhotoGallery.kt
package com.nathanhaze.gifcreator.gallery

import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import androidx.core.app.NavUtils
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.nathanhaze.gifcreator.R
import com.nathanhaze.gifcreator.gallery.GalleryFragment


class PhotoGallery : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.fragment_photo_gallery)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(android.R.id.content)) { v, insets ->
            val bars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(bars.left, bars.top, bars.right, bars.bottom)
            insets
        }
        com.nathanhaze.gifcreator.manager.Utils.trackScreenView(this, "Gallery")
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.frame_main, GalleryFragment())
        ft.commit()

    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                NavUtils.navigateUpFromSameTask(this)
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}